### PR TITLE
Make label for ilastik executable system specific

### PR DESCRIPTION
**Pull request by m-novikov**
Closes: #54

### CODE REVIEW

**Comment by k-dominik** [comment]
this could potentially also address #57 :)

File: src/main/java/org/ilastik/ilastik4ij/ui/IlastikOptions.java [hunk 1]
```diff
@@ -17,10 +18,11 @@
  */
 @Plugin(type = OptionsPlugin.class, menuPath = "Plugins>ilastik>Configure ilastik executable location")
 public class IlastikOptions extends OptionsPlugin {
-    @Parameter(label = "Path to ilastik executable, e.g. " +
-            "MacOS: /Applications/ilastik-1.3.3-OSX.app/Contents/MacOS/ilastik, " +
-            "Windows: C:\\\\Program Files\\ilastik\\ilastik.exe, " +
-            "Linux: /opt/ilastik/run_ilastik.sh")
+    private static final String ILASTIK_PATH_WIN = "C:\\\\Program Files\\ilastik\\ilastik.exe";
+    private static final String ILASTIK_PATH_LINUX = "/opt/ilastik/run_ilastik.sh";
+    private static final String ILASTIK_PATH_MACOS = "/Applications/ilastik-1.3.3-OSX.app/Contents/MacOS/ilastik";
```
Comment (k-dominik) [inline src/main/java/org/ilastik/ilastik4ij/ui/IlastikOptions.java:23]: My only suggestion here would be to add some versions to the paths in windows and macos, and make clear that those are examples (as in the original code, it said "e.g. ..."

Comment (emilmelnikov) [reply src/main/java/org/ilastik/ilastik4ij/ui/IlastikOptions.java:23]:
```java
private static final String[][] EXAMPLE_PATHS = {
    {"win", "C:\\\\Program Files\\ilastik\\ilastik.exe"},
    {"linux", "/opt/ilastik/run_ilastik.sh"},
    {"mac", "/Applications/ilastik-1.3.3-OSX.app/Contents/MacOS/ilastik"},
};

// ...

String path = null;
for (String[] e : EXAMPLE_PATHS) {
    if (os.contains(e[0]) {
        path = e[1];
        break;
    }
}

if (path == null) {
    throw new RuntimeException("Unknown os " + os);
}

getInfo().getMutableInput("executableFile", File.class).setLabel("Path to ilastik executable, e.g. " + path);
```

Comment (m-novikov) [reply src/main/java/org/ilastik/ilastik4ij/ui/IlastikOptions.java:23]: > make clear that those are examples (as in the original code, it said "e.g. ..."

It already adds "e.g." if the system is identified. See [highlight](https://github.com/ilastik/ilastik4ij/pull/55/files#diff-e2906d73755dc662754edb45de0fc994R49)

Comment (k-dominik) [reply src/main/java/org/ilastik/ilastik4ij/ui/IlastikOptions.java]: > > make clear that those are examples (as in the original code, it said "e.g. ..."
> 
> It already adds "e.g." if the system is identified. See [highlight](https://github.com/ilastik/ilastik4ij/pull/55/files#diff-e2906d73755dc662754edb45de0fc994R49)

sorry, missed that :)

**Comment by m-novikov** [comment]
> @emilmelnikov mentioned that using the old imagej1 api to query for OS should probably be avoided.

I would use it because:
1. There is no alternative in imagej2 API
2. There is no deprecation annotation
3. Personally I doubt that it will ever be deprecated

**Comment by k-dominik** [comment]
> * There is no alternative in imagej2 API
> 
> * There is no deprecation annotation
> 
> * Personally I doubt that it will ever be deprecated

compelling argument :) @emilmelnikov ?

**Comment by emilmelnikov** [comment]
I looked up the implementation of `IJ.isOSX()` and it just checks a system property. So we could just replace it with:

```java
bool isMac = System.getProperty("os.name").toLowerCase().startsWith("mac");
```

**Comment by m-novikov** [comment]
> I looked up the implementation of `IJ.isOSX()` and it just checks a system property. So we could just replace it with:
> 
> ```java
> bool isMac = System.getProperty("os.name").toLowerCase().startsWith("mac");
> ```

I mean why would we implement it ourselves if it's already implemented for us? :)

**Comment by emilmelnikov** [comment]
> I mean why would we implement it ourselves if it's already implemented for us? :)

I don't have a strong opinion about this, but I wouldn't using ImageJ 1 for such a trivial thing.

**Comment by k-dominik** [comment]
At least it's going to be easy to replace if they every get rid of `IJ` :)

**Pull request merged**